include/wx/filedlg.h

```diff
@@ -104,10 +104,19 @@ class WXDLLIMPEXP_CORE wxFileDialogBase: public wxDialog
     virtual void SetFilterIndex(int filterIndex) { m_filterIndex = filterIndex; }
 
     virtual wxString GetMessage() const { return m_message; }
-    virtual wxString GetPath() const { return m_path; }
+    virtual wxString GetPath() const
+    {
+        wxCHECK_MSG( !HasFlag(wxFD_MULTIPLE), wxEmptyString, wxT("GetPath called when wxFD_MULTIPLE defined") );
```

Comment:
I'd like to change a couple of things here and make this:
```suggestion
        wxASSERT_MSG( !HasFlag(wxFD_MULTIPLE), "When using wxFD_MULTIPLE, must call GetPaths() instead" );
```

The changes are:

1. Use `ASSERT` because this is less code (both source and binary) than `CHECK` while the effect is the same.
1. Don't use `wxT()`: it shouldn't be used in any new code (i.e. written since wx 2.9.1 or thereabouts). If you care about not doing conversions during run-time, use `wxS()`, if you don't (and here it really doesn't matter), don't use any macros at all.
1. Last but the most important: use prescriptive error message indicating what must be done to fix the problem. Experience shows that while this may seem obvious to you and me, most users have no idea what do they need to do when encountering such a message and are probably much more likely to remove `wxFD_MULTIPLE` than to actually fix the problem.

Reply:
Regarding `wxASSERT` vs `wxCHECK`. We already talked about it and agreed that an empty string should be returned?

Please see [here](https://trac.wxwidgets.org/ticket/18736#comment:17) for my explanation of how the multiple selected items are returned on MSW and your reply in the first paragraph of the next comment there.

Reply:
Yes, I had thought that we had agreed on returning the empty string if called with multiple selection enabled - that was why I used the check message instead of an assert. I guess we could do an assert and then do

```
wxASSERT_MSG( !HasFlag(wxFD_MULTIPLE), "When using wxFD_MULTIPLE, must call GetPaths() instead" );
if ( HasFlag(wxFD_MULTIPLE) )
    return wxEmptyString;
```
but I thought the check was cleaner.

Reply:
But I thought that `m_path` would be empty anyhow here, when `wxFD_MULTIPLE` is on, wouldn't it?

Anyhow, if you prefer to use `wxCHECK` it's not a big deal (the points are actually in the order of increasing importance on something close to log scale), but let's please use `wxString()` rather than `wxEmptyString` in this case (the latter is again something that has no reason to be used in the new code).

Reply:
At least on GTK, calling `GetPath` when `wxFD_MULTIPLE` is defined still returns a non-empty string - so I decided to make all ports the same and explicitly return an empty string. That is also safer, because if some logic populating the `m_path` variable changes down the road, it doesn't change the return value here. You can see that in the dialogs sample with this patch:

```
diff --git a/samples/dialogs/dialogs.cpp b/samples/dialogs/dialogs.cpp
index c5c1cc1da7..934a57e57a 100644
--- a/samples/dialogs/dialogs.cpp
+++ b/samples/dialogs/dialogs.cpp
@@ -1661,8 +1661,11 @@ void MyFrame::FilesOpen(wxCommandEvent& WXUNUSED(event) )
     {
         wxArrayString paths, filenames;
 
-        dialog.GetPaths(paths);
-        dialog.GetFilenames(filenames);
+        paths.Add( dialog.GetPath() );
+        filenames.Add( dialog.GetFilename() );
+
+        //dialog.GetPaths(paths);
+        //dialog.GetFilenames(filenames);
 
         wxString msg, s;
         size_t count = paths.GetCount();
```

Reply:
I see, thanks for the explanation and sorry for a wrong comment. No problem with keeping `wxCHECK()` if we really need it, of course.

Reply:
I  have updated this in the most recent commit.